tests/Models/Multiplayer/RoomTest.php

```diff
@@ -15,6 +15,39 @@
 
 class RoomTest extends TestCase
 {
+    /**
+     * @dataProvider startGameDurationDataProvider
+     */
+    public function testStartGameDuration(int $duration, bool $isSupporter, ?string $errorMessageKey)
+    {
+        $beatmap = Beatmap::factory()->create();
+        $user = User::factory();
+        if ($isSupporter) {
+            $user = $user->supporter();
+        }
+
+        $user = $user->create();
+
+        $params = [
+            'duration' => $duration,
+            'name' => 'test',
+            'playlist' => [
+                [
+                    'beatmap_id' => $beatmap->getKey(),
+                    'ruleset_id' => $beatmap->playmode,
+                ],
+            ],
+        ];
+
+        if ($errorMessageKey !== null) {
+            $this->expectException(InvariantException::class);
+            $this->expectExceptionMessage(osu_trans($errorMessageKey));
+        }
+
+        $room = (new Room())->startGame($user, $params);
+        $this->assertSame($errorMessageKey === null, $room->exists);
```

Comment:
that's an interesting combination between `$this->assertTrue($room->exists)` assertion and `if ($errorMessageKey === null) { ... }` condition check (which is the opposite of the previous check above).

At a glance it looks like it's checking `false` if `$errorMessageKey` is not null except it'll never happen (as the `startGame` call throws)